xonsh/tools.py

```diff
@@ -42,6 +42,13 @@
                             ON_LINUX, ON_WINDOWS, PYTHON_VERSION_INFO)
 
 
+# regular expressions for matching enviroment variables
+# i.e $FOO, ${'FOO'}
+posix_envvar_regex = re.compile(r'\$({(?P<quote>[\'\"])|)(?P<envvar>\w+)((?P=quote)}|(?:\1\b))')
```

Comment:
You should make these lazy. See the concept here: https://github.com/laerus/xonsh/blob/2dc7f2b5c322ddb640eacd7d814c48371cc6c445/xonsh/tools.py#L213